h2o-py/h2o/frame.py

```diff
@@ -509,7 +509,10 @@ def logical_negation(self):
 
 
     def _unop(self, op):
-        return H2OFrame._expr(expr=ExprNode(op, self), cache=self._ex._cache)
+        ret = H2OFrame._expr(expr=ExprNode(op, self), cache=self._ex._cache)
+        if ret._ex._cache._names is not None:
```

Comment:
too much indirection with `ret._ex._cache._names`. What is the intent ? I sense this is an expression evaluation that has some memoization but is not clear why all that name mangling.